torch/_dynamo/guards.py

```diff
@@ -176,6 +179,7 @@ def __init__(
         # Code is python expression strings generated for each guard
         self.code: List[str] = []
         self.tensor_check_names = []
+        self.tensor_check_ids = {}
```

Comment:
@voznesenskym informed me in private communication that we do not actually check the id()s of tensors in any guards. So this leaves me puzzled as to what this dictionary is actually doing... based on the name, it sure looks like a list of tensors to check the ids of.

Reply:
I added a lot of comments about how the mapping works, see updated source.

Reply:
> The testing is really insufficient (aka there aren't any tests)

Discussed offline. All dynamo tests run w/ dynamic shapes.

That being said, I plan to add some more tests here.